circleci: failed tests (14796): TestTxnRestart

The following test appears to have failed:

[#14796](https://circleci.com/gh/cockroachdb/cockroach/14796):

```
I160318 15:28:21.301248 storage/replica.go:1274  reproposing 1 commands after empty entry
I160318 15:28:21.301312 storage/replica.go:1274  reproposing 1 commands after empty entry
I160318 15:28:21.301780 storage/replica.go:1565  found sequence cache entry for 5bb1d676@epo=0,seq=15: epo=0,seq=15
E160318 15:28:21.301947 storage/replica.go:1358  [node=1,store=1,range=2] error executing raft command: retry txn "sql/lease.go:154 LeaseStore.Acquire" id=5bb1d676 key=/Table/11/1/51/1/Fri Mar 18 15:33:20 UTC 2016/1/0 rw=false pri=0.07496239 iso=SERIALIZABLE stat=PENDING epo=0 ts=1458314901.299416735,0 orig=1458314901.299416735,0 max=1458314901.299416735,0
I160318 15:28:21.302336 storage/replica_command.go:1056  failed to push "storage/replica_command.go:1624 (*Replica).AdminSplit" id=12deeafd key=/Local/Range/"\xba"/RangeDescriptor rw=false pri=0.02397508 iso=SERIALIZABLE stat=PENDING epo=0 ts=1458314901.277781063,0 orig=1458314901.277781063,0 max=1458314901.277781063,0
panic: interface conversion: interface is nil, not error

goroutine 23364 [running]:
panic(0x16b4940, 0xc8205caec0)
	/usr/local/go/src/runtime/panic.go:464 +0x3e6
github.com/cockroachdb/cockroach/testutils/storageutils.(*ReplayProtectionFilterWrapper).run(0xc820659ef0, 0x7f783be12550, 0xc820a2e9f0, 0xc820048698, 0x8, 0x0, 0x1, 0x7f783be1a898, 0xc820524640, 0x143cf9ad1e7cbbcb, ...)
	/go/src/github.com/cockroachdb/cockroach/testutils/storageutils/mocking.go:95 +0x1d8
github.com/cockroachdb/cockroach/testutils/storageutils.(*ReplayProtectionFilterWrapper).(github.com/cockroachdb/cockroach/testutils/storageutils.run)-fm(0x7f783be12550, 0xc820a2e9f0, 0xc820048698, 0x8, 0x0, 0x1, 0x7f783be1a898, 0xc820524640, 0x143cf9ad1e7cbbcb, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/testutils/storageutils/mocking.go:77 +0x4e
github.com/cockroachdb/cockroach/sql_test.(*CommandFilters).runFilters(0xc820216000, 0x7f783be12550, 0xc820a2e9f0, 0xc820048698, 0x8, 0x0, 0x1, 0x7f783be1a898, 0xc820524640, 0x143cf9ad1e7cbbcb, ...)
	/go/src/github.com/cockroachdb/cockroach/sql/main_test.go:70 +0xb3
github.com/cockroachdb/cockroach/sql_test.(*CommandFilters).(github.com/cockroachdb/cockroach/sql_test.runFilters)-fm(0x7f783be12550, 0xc820a2e9f0, 0xc820048698, 0x8, 0x0, 0x1, 0x7f783be1a898, 0xc820524640, 0x143cf9ad1e7cbbcb, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/sql/main_test.go:189 +0x4e
github.com/cockroachdb/cockroach/storage.(*Replica).executeCmd(0xc820bcad20, 0x7f783be12550, 0xc820a2e9f0, 0xc820048698, 0x8, 0x0, 0x7f783be120b0, 0xc820a2ea20, 0xc820773018, 0x143cf9ad1e7cbbcb, ...)
--
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:1807 +0x35a
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8204c6000, 0xc820655000)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62
FAIL	github.com/cockroachdb/cockroach/sql	42.243s
=== RUN   TestEval
--- PASS: TestEval (0.01s)
=== RUN   TestEvalError
--- PASS: TestEvalError (0.00s)
=== RUN   TestEvalComparisonExprCaching
--- PASS: TestEvalComparisonExprCaching (0.00s)
=== RUN   TestSimilarEscape
--- PASS: TestSimilarEscape (0.00s)
=== RUN   TestQualifiedNameString
--- PASS: TestQualifiedNameString (0.00s)

```
Please assign, take a look and update the issue accordingly.

--- a/testutils/storageutils/mocking.go
+++ b/testutils/storageutils/mocking.go
@@ -25,10 +25,6 @@ import (
 	"github.com/cockroachdb/cockroach/storage/storagebase"
 )
 
-// savedError silences the linter from complaining when detectReplayLocks wants
-// to return an error as the first (as opposed to the last) return value.
-type savedError error
-
 // FilterArgs groups the arguments to a ReplicaCommandFilter.
 type FilterArgs struct {
 	Ctx   context.Context
@@ -62,7 +58,7 @@ type ReplicaCommandFilter func(args FilterArgs) error
 // from Raft replays.
 type ReplayProtectionFilterWrapper struct {
 	sync.RWMutex
-	processedCommands map[raftCmdIDAndIndex]savedError
+	processedCommands map[raftCmdIDAndIndex]error
 	filter            ReplicaCommandFilter
 }
 
@@ -71,17 +67,10 @@ type ReplayProtectionFilterWrapper struct {
 func WrapFilterForReplayProtection(
 	filter ReplicaCommandFilter) ReplicaCommandFilter {
 	wrapper := ReplayProtectionFilterWrapper{
-		processedCommands: make(map[raftCmdIDAndIndex]savedError),
+		processedCommands: make(map[raftCmdIDAndIndex]error),
 		filter:            filter,
 	}
 	return wrapper.run
-}
-
-func (c *ReplayProtectionFilterWrapper) detectReplayLocked(
-	cmdID storagebase.CmdIDKey, index int) (savedError, bool) {
-	err, found := c.processedCommands[raftCmdIDAndIndex{cmdID, index}]
-	// We've detected a replay.
-	return err, found
 }
 
 // run executes the wrapped filter.
@@ -91,15 +80,17 @@ func (c *ReplayProtectionFilterWrapper) run(args FilterArgs) error {
 	defer c.RUnlock()
 
 	if args.InRaftCmd() {
-		if err, found := c.detectReplayLocked(args.CmdID, args.Index); found {
-			return err.(error)
+		if err, found :=
+			c.processedCommands[raftCmdIDAndIndex{args.CmdID, args.Index}]; found {
+			// We've detected a replay.
+			return err
 		}
 	}
 
 	err := c.filter(args)
 
 	if args.InRaftCmd() {
-		c.processedCommands[raftCmdIDAndIndex{args.CmdID, args.Index}] = savedError(err)
+		c.processedCommands[raftCmdIDAndIndex{args.CmdID, args.Index}] = err
 	}
 
 	return err

